src/parser.cpp

```diff
@@ -539,19 +540,22 @@ result_type parse_url(std::string_view user_input,
       }
       case ada::state::QUERY: {
         ada_log("QUERY ", helpers::substring(url_data, input_position));
-        // Let queryPercentEncodeSet be the special-query percent-encode set if
-        // url is special; otherwise the query percent-encode set.
-        const uint8_t* query_percent_encode_set =
-            url.is_special() ? ada::character_sets::SPECIAL_QUERY_PERCENT_ENCODE
-                             : ada::character_sets::QUERY_PERCENT_ENCODE;
-
-        // Percent-encode after encoding, with encoding, buffer, and
-        // queryPercentEncodeSet, and append the result to url's query.
-        url.update_base_search(helpers::substring(url_data, input_position),
-                               query_percent_encode_set);
-        ada_log("QUERY update_base_search completed ");
-        if (fragment.has_value()) {
-          url.update_unencoded_base_hash(*fragment);
+        if (store_values) {
```

Comment:
```suggestion
        if constexpr (store_values) {
```

Reply:
I just noticed and understood this comment, nice! I noticed you mentioned the `update_unencoded_base_hash`, and I'd like to confirm some other cases. For instance, `update_base_pathname` and `update_base_search` would be methods we want to avoid for the can_parse as well, right?

```
if constexpr (result_type_is_ada_url) {
  url.path = base_url->path;
  url.query = base_url->query;
} else {
  url.update_base_pathname(base_url->get_pathname());
  url.update_base_search(base_url->get_search());
}
url.update_unencoded_base_hash(*fragment);
return url;
```